accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/ElementIterator.java

```diff
@@ -112,10 +112,12 @@ private boolean processKeyValue(Key key, Text columnFamily, Value value) {
         }
 
         if (getVisibilitySignal().equals(columnFamily)) {
-            elementData.visibility = key.getColumnVisibility();
-            elementData.timestamp = key.getTimestamp();
-            processSignalColumn(key.getColumnQualifier());
-            return true;
+            if (elementData.timestamp == 0 || key.getTimestamp() > elementData.timestamp) {
```

Comment:
When is timestamp 0?

Reply:
`elementData.timestamp` starts off as 0 before it is set the first time. We probably don't need that extra check since the second half of the conditional will catch it. I'll remove it and push the change.

Reply:
I cleaned up the if statement